### PR TITLE
Support header with embedded delimiter in csv writer

### LINKED ISSUE #8713: [BUG] `to_csv` doesn't quote column titles containing commas
Minimal reproducible example:
```python
import cudf
import pandas as pd
import numpy as np

a_pd = pd.DataFrame([[1, 2]], columns=["A", "B,C"])
a_cudf = cudf.from_pandas(a_pd)

a_pd.to_csv("a_pd.csv")
a_cudf.to_csv("a_cudf.csv")

with open("a_pd.csv", 'r') as f:
    print(f.read())
with open("a_cudf.csv", 'r') as f:
    print(f.read())
```

Output:
```
,A,"B,C"
0,1,2

,A,B,C
0,1,2
```

Note that the csv file generated by cudf has 4 columns in the header and 3 columns in the data.

### PR DESCRIPTION
**Pull request by davidwendt**
Closes #8713 

If the column delimiter (e.g. comma) appears in a column header, then the header string must be enclosed in double-quotes when written to the output csv file. The logic that writes the header line is updated in this PR to handle an embedded delimiter, terminator, or double-quote character. Also added a gtest with a comma inside a header string.

### CODE REVIEW

**Review by codereport — APPROVED** [review]
lgtm :+1:

File: cpp/tests/io/csv_test.cpp [hunk 1]
```diff
@@ -1845,6 +1845,28 @@ TEST_F(CsvReaderTest, StringsEmbeddedDelimiter)
   CUDF_TEST_EXPECT_TABLES_EQUIVALENT(input_table, result.tbl->view());
 }
 
+TEST_F(CsvReaderTest, HeaderEmbeddedDelimiter)
+{
+  std::vector<std::string> names{"header1", "header,2", "quote\"embedded", "\"quoted\""};
```
Comment (robertmaynard) [inline cpp/tests/io/csv_test.cpp:1850]: Should we have a test case with the `terminator` embedded into it?

**Review by robertmaynard — APPROVED** [review]

**Review by robertmaynard — APPROVED** [review]
LGTM

**Comment by davidwendt** [comment]
@gpucibot merge

**Pull request merged**